src/test/java/com/academia/andruhovich/library/controller/AuthorControllerTest.java

```diff
@@ -55,62 +51,78 @@
     @WithMockUser(username = "admin_mock", roles = "USER", authorities = AUTHORITY_READ, password = "12356")
     @Test
     void getAuthors() throws Exception {
-        mockMvc.perform(MockMvcRequestBuilders
-                .get("/api/authors")
-                .accept(APPLICATION_JSON))
+        String response = mockMvc.perform(MockMvcRequestBuilders
+                        .get("/api/authors")
+                        .accept(APPLICATION_JSON))
                 .andDo(print())
                 .andExpect(status().isOk())
-                .andExpect(MockMvcResultMatchers.jsonPath("$").exists())
-                .andExpect(MockMvcResultMatchers.jsonPath("$[*].id").isNotEmpty());
+                .andReturn().getResponse().getContentAsString();
+
+        List<AuthorDto> authors = getAuthorDtos(response);
```

Comment:
Have you tried to deserialize directly on List<AuthorDto>?

Reply:
Refactored it